lib/logstash/outputs/elasticsearch.rb

```diff
@@ -557,9 +554,30 @@ def common_event_params(event)
     #      }
     params[:pipeline] = target_pipeline unless (target_pipeline.nil? || target_pipeline.empty?)
 
+    params[:version] = resolve_version(event, event_version)
+    params[:version_type] = resolve_version_type(event, event_version_type)
```

Comment:
_alternate_ approaches to `Hash#compact` to avoid storing nil values:

1. use a local variable:
    ```suggestion
        resolved_version_type = resolve_version_type(event, event_version_type)
        params[:version_type] = resolved_version_type unless resolved_version_type.nil?
    ```

2. rely on safe-descent `&.` with `Object#tap`:
    ```suggestion
        resolve_version_type(event, event_version_type)&.tap { |resolved| params.store(:version_type, resolved) }
    ```

Reply:
Thank's @yaauie to have provided also this alternative, I've opted for the one-liner `compact`

Reply:
`routing => nil` is expected in some tests: 
https://app.travis-ci.com/github/logstash-plugins/logstash-output-elasticsearch/jobs/613384063#L4748

This comes from the fixture

https://github.com/logstash-plugins/logstash-output-elasticsearch/blob/47a5169a464ef637539e0dfffbd595bb3ddd4a86/spec/integration/outputs/retry_spec.rb#L9-L19

Reply:
I've reverted the behaviour only for `routing` with commit e91210c084d6e494e7df3d7011743947d6205b73